### DISCUSSION TITLE
how can i renew my certificate automatically!!! using script

### DISCUSSION THREAD

**Discussion by sandraimmaculate** [Security-Onion-Solutions/securityonion#12790] [Unsupported Versions] [answered]
hi 
https://gist.github.com/l00ps/5753fa7fe55f6757fcc088506637a13e i'm going through this they have been mentioned like 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/73ea8285-6180-453f-9ff5-399eb8c314fa)
i followed those soon after doing those change my nginx container is missing, so i revert those and restart the nginx. then it started working.

then i installed certboot created a path for webroot  and executed the command 
sudo certbot certonly --webroot -w /usr/share/nginx/letsencryptbase -d so.#####.com -v
im getting this error 
sudo certbot certonly --webroot -w /usr/share/nginx/letsencryptbase -d so.######.com
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Requesting a certificate for so.####.com

Certbot failed to authenticate some domains (authenticator: webroot). The Certificate Authority reported these problems:
  Domain: so.####.com
  Type:   connection
  Detail: *.*.*.*: Fetching http://so.####.com/.well-known/acme-challenge/1yrjzvJRXdeDmw31DL: Timeout during connect (likely firewall problem)

Hint: The Certificate Authority failed to download the temporary challenge files created by Certbot. Ensure that the listed domains serve their content from the provided --webroot-path/-w and that files created there can be downloaded from the internet.

Some challenges have failed.

i have added the port 80 in iptables.

i think even though i have added those ports in iptables, it not opened i guess because if i access from another system i'm not able to security onion server through 80.
even i have tried with standalone plugin i'm facing the same issue there also.

if i followed the bellow steps which is mentioned in security onion document do i can create a crontab for automatic renewal when certificate got expired. if there please explain
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/342729f3-7c17-4102-bd61-15c18750b79b)

**Comment by sandraimmaculate** [comment]
if i want to open any  ports means how can i do that 
how can i open port 80 not only for specific ip but for all

**Comment by petiepooo** [comment]
That is an advanced configuration setting.  See https://docs.securityonion.net/en/2.4/firewall.html#creating-a-custom-host-group-with-a-custom-port-group

**Reply by sandraimmaculate** [reply]
this i have tried, this will work for opening the port for specific ip only !

**Comment by TOoSmOotH** [comment]
Please please please do not open your SO to the internet. Use the DNS challenge to get your cert then drop it into the proper place in the config.

**Reply by sandraimmaculate** [reply]
except DNS challenge do we have any other option! ??  Because we are using PDNS. we have to renew the certificate manually whenever it get expired right.

do we can automate it by running the script!
how can we do that

**Reply by sandraimmaculate** [reply]
??

**Reply by sandraimmaculate** [reply]
hello please please do reply for this 

> except DNS challenge do we have any other option! ?? Because we are using PDNS. we have to renew the certificate manually whenever it get expired right.
> 
> do we can automate it by running the script! how can we do that

Now it going to expire now i want to run a renewal script so only for that time when the script is running i want to open the port 80 for 0.0.0.0/0 and again i will be removed soon after the script completion

**Reply by sandraimmaculate** [reply]
please suggest!! it been 3 months i'm asking about this

**Reply by dougburks** [reply, marked as answer]
We responded back on April 16:
https://github.com/Security-Onion-Solutions/securityonion/discussions/12790#discussioncomment-9135462
> Please please please do not open your SO to the internet.

Our official recommendation is that you find some other way to accomplish your goal without opening your Security Onion instance to the Internet.
